vet-service/src/main/java/com/petclinic/vet/util/EntityDtoUtil.java

```diff
@@ -0,0 +1,109 @@
+package com.petclinic.vet.util;
+/**
+ @author Kamilah Hatteea & Brandon Levis : Vet-Service
+  * Worked together with (Code with Friends) on IntelliJ IDEA
+  * <p>
+  * User: @Kamilah Hatteea
+  * Date: 2022-09-22
+  * Ticket: feat(VVS-CPC-554): edit veterinarian
+  * User: Brandon Levis
+  * Date: 202
+  * Ticket: feat(VVS-CPC-553): add veterinarian
+ */
+
+import com.petclinic.vet.dataaccesslayer.Specialty;
+import com.petclinic.vet.dataaccesslayer.Vet;
+import com.petclinic.vet.exceptions.InvalidInputException;
+import com.petclinic.vet.servicelayer.SpecialtyDTO;
+import com.petclinic.vet.servicelayer.VetDTO;
+import lombok.Generated;
+import org.springframework.beans.BeanUtils;
+
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+
+public class EntityDtoUtil {
+    @Generated
+    public EntityDtoUtil(){}
+
+    public static VetDTO toDTO(Vet vet) {
+        VetDTO dto = new VetDTO();
+        dto.setVetId(vet.getVetId());
+        dto.setFirstName(vet.getFirstName());
+        dto.setLastName(vet.getLastName());
+        dto.setEmail(vet.getEmail());
+        dto.setPhoneNumber(vet.getPhoneNumber());
+        dto.setImage(vet.getImage());
+        dto.setResume(vet.getResume());
+        dto.setWorkday(vet.getWorkday());
+        dto.setActive(vet.isActive());
+        dto.setSpecialties(toDTOSet(vet.getSpecialties()));
+        return dto;
+    }
+
+    public static Vet toEntity(VetDTO dto) {
+        Vet vet = new Vet();
+        vet.setVetId(generateVetId());
+        vet.setFirstName(dto.getFirstName());
+        vet.setLastName(dto.getLastName());
+        vet.setEmail(dto.getEmail());
+        vet.setPhoneNumber(dto.getPhoneNumber());
+        vet.setImage(dto.getImage());
+        vet.setResume(dto.getResume());
+        vet.setWorkday(dto.getWorkday());
+        vet.setActive(dto.isActive());
+        vet.setSpecialties(toEntitySet(dto.getSpecialties()));
+        return vet;
+    }
+
+    public static String generateVetId() {
+        Random random = new Random();
+        int number = random.nextInt(99999);
+        return "22" + (String.format("%05d", number));
+    }
+
+    public static SpecialtyDTO toDTO(Specialty specialty) {
+        SpecialtyDTO dto = new SpecialtyDTO();
+        BeanUtils.copyProperties(specialty, dto);
+        return dto;
+    }
+
+    public static Specialty toEntity(SpecialtyDTO dto) {
+        Specialty specialty = new Specialty();
+        BeanUtils.copyProperties(dto, specialty);
+        return specialty;
+    }
+
+    public static Set<SpecialtyDTO> toDTOSet(Set<Specialty> specialties) {
+        Set<SpecialtyDTO> specialtyDTOS = new HashSet<>();
+        for (Specialty specialty:
+             specialties) {
+            SpecialtyDTO specialtyDTO = toDTO(specialty);
+            specialtyDTOS.add(specialtyDTO);
+        }
+
+        return specialtyDTOS;
+    }
+
+    public static Set<Specialty>  toEntitySet(Set<SpecialtyDTO> specialtyDTOS){
```

Comment:
Little typo here, missing a space between the parenthesis and the bracket, making it different from the rest of the methods in the file.